projects/module-2/whats-cookin-part-one.md

```diff
@@ -0,0 +1,277 @@
+You as the developer will need to make decisions about the shape of data you'll want back from various functions.  Consider how you will be using this data to update the DOM. 
+</section>
+
+#### User Stories (Dashboard)
+
+Use the `domUpdates.js` file to add information to the DOM. This JS file should call functions from `index.js`. There should not be any DOM manipulation within your `index.js` file.
+
+- As a user, I should be able to view all recipes.
+- As a user, I should be able to click on a recipe to view more information including directions, ingredients needed, and total cost.
+- As a user, I should be able to filter recipes by a tag. (Extension option: by multiple tags)
+- As a user, I should be able to search recipes by their name. (Extension option: by name or ingredients)
+
+---
+
+### Iteration 2 - Users
+
+#### User Data
+
+You should have functions that:
+- Allow a user to add/remove a recipe to/from their recipesToCook list (add to my `recipesToCook`)
+- Filter my `recipesToCook` by a tag. (Extension option: filter by multiple tags)
+- Filter my `recipesToCook` by its name. (Extension option: filter by name or ingredients)
+
+<!-- The last two functions above should be the same as 2 they already built, they shouldn't build a duplicate -->
+
+#### User Stories
+
+*On load, a user should be chosen at random.*
+
+* As a user, I should be able to add/remove a recipe to a list of recipes to cook
+* As a user, I should be able to filter my `toCook` recipes by a tag. (Extension option: filter by multiple tags)
+* As a user, I should be able to search my `toCook` recipes by its name. (Extension option: search by name or ingredients)
+
+
+---
+
+### Iteration 3 - Fetch Calls
+
+You will no longer be receiving your data from a hardcoded data file, but rather implementing the fetch API for accessing the data from provided Endpoints in the table below.
+
+<section class="note">
+### Hint
+
+The expectation for Mod 2 is that you will avoid using `async/await`.  We know `async/await` is tempting, but it is important that you are able to work with the approaches that pre-date the introduction of `async/await`.  Consider doing some research on [Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all).
+</section>  
+
+| Description | URL | Method | Required Properties for Request | Sample Successful Response |
+|----------|-----|--------|---------------------|-----------------|
+| Get all users |`https://what-s-cookin-starter-kit.herokuapp.com/api/v1/users`| GET | none | An array containing all users |
+| Get all ingredients |`https://what-s-cookin-starter-kit.herokuapp.com/api/v1/ingredients` | GET | none | An array containing all ingredients |
+| Get all recipes | `https://what-s-cookin-starter-kit.herokuapp.com/api/v1/recipes` | GET | none | An array containing all recipes |
+
+Once you've got fetch working on all three arrays, you can delete your hardcoded data file!
+
+
+### Extensions
```

Comment:
Is there a way we can call this something other than "Extensions", since every other FE project at Turing uses "extension" to indicate something they do not have to do?

Maybe we could call it "Additional Feature" or "Choose Your Own Adventure" or something.

(The language would be updated on line 199, too)

Reply:
I thought that these are also intended to be extensions that students don't have to do.  If they did them, it would earn them a WOW on functionality but they aren't requirements for a YES.   Are these expectations or extensions currently?  I'll post in foo-mod2 to see what the current M2 instructors think.

Reply:
But I see that the note to `Choose at least one` does make it seem like they are expected.  Hmmm